### ISSUE TITLE
Pagination events

### ISSUE THREAD

**Issue by ahmadhsalim** [daniel-nagy/md-data-table#14]
I think the mdDataTablePagination directive should have an ng-change event, which fires whenever a page is changed. Meaning whenever any of the pagination buttons are clicked.

**Comment by daniel-nagy** [comment]
How about,

```javascript
// my controller
$scope.onPageChange = function (page) {
  // do stuff
}
```

```html
<md-data-table-toolbar layout-align="end">
  <md-data-table-pagination on-change="onPageChange"></md-data-table-pagination>
</md-data-table-toolbar>
```

I had this implemented at one point then discarded it during a refactor but I could reimplement it again. I agree this would be nice so you wouldn't have to register `$watch` listeners. Would you care to have the same functionality for the `md-filter` (will be `md-order` in the next release for better naming convention) attribute?

**Comment by ahmadhsalim** [comment]
I think components like md-data-table should have lot of event handlers for everything it does, even `md-filter` (`md-order`). The example shown above is good. Perhaps the parameters should be, `page` and `limit`.

**Comment by daniel-nagy** [comment]
Okay, this is in spot two of the queue behind row selection backgrounds :+1:

**Comment by ahmadhsalim** [comment]
Can you make it so that the pagination and order change event is NOT run when the page is loaded? Instead it should be run only when the page is changed by the user interactively. I'm having some trouble because the page change event is run when the page is first loaded.
Have a look at the `<pagination>` directive in this [link](https://angular-ui.github.io/bootstrap/#/pagination). I use it in a similar way, but I cannot use the same codes for this because of the event being fired every time the state is loaded.

**Comment by ahmadhsalim** [comment]
Also I don't understand how you have implemented it in your `nutritionController`. Where are the `onOrderChange` and `onPageChange` event handlers??

**Comment by daniel-nagy** [comment]
I'll admit that this needs more work and the reason the function gets triggered on page load is because the `ng-change` directive automatically fires in the select dropdown. I'll revisit this after I finish implementing the `md-disable-select` attribute.

> "Also I don't understand how you have implemented it in your nutritionController. Where are the onOrderChange and onPageChange event handlers??"

I think this is just a mistake on my part. I should of removed the attributes in the template as `ng-repeat` is smart enough to know when the values change.

**Comment by daniel-nagy** [comment]
@ahmadhsalim I'm not positive but I think master branch of Angular Material contains a fix for `ngChange` firing immediately inside `mdSelect`. I've pulled down master branch for my current project and it is no longer firing on page load. It looks like they've made some other changes to the template so the styling is not completely accurate but is still very usable.

Now that I have finished `md-disable-select` I'm going to revisit this. I'd like to see if i can wrap the function that gets passed to `mdTrigger` in a promise API so even if the developer calls it in their template a loading indicator will be displayed.
